src/ast.ts

```diff
@@ -56,12 +71,28 @@ const startNoDot = '(?!\\.)'
 // characters that indicate a start of pattern needs the "no dots" bit,
 // because a dot *might* be matched. ( is not in the list, because in
 // the case of a child extglob, it will handle the prevention itself.
-const addPatternStart = new Set(['[', '.'])
+const addPatternStart = new SafeSet(['[', '.'])
 // cases where traversal is A-OK, no dot prevention needed
-const justDots = new Set(['..', '.'])
-const reSpecials = new Set('().*{}+?[]^$\\!')
+const justDots = new SafeSet(['..', '.'])
+const reSpecials = new SafeSet([
+  "'",
+  '(',
+  ')',
+  '.',
+  '*',
+  '{',
+  '}',
+  '+',
+  '?',
+  '[',
+  ']',
+  '^',
+  '$',
+  '\\',
+  '!',
+])
```

Comment:
Does `SafeSet` only take arrays, not arbitrary iterables?

Reply:
see the implementation here https://github.com/isaacs/node-primordials/pull/9/files# I guess it would work with any iterable, but didn't try